src/Domain/Orientation.cpp

```diff
@@ -53,6 +53,20 @@ std::array<SegmentId, VolumeDim> Orientation<VolumeDim>::mapped(
   return result;
 }
 
+template <size_t VolumeDim>
+Orientation<VolumeDim> Orientation<VolumeDim>::inverse() const noexcept {
+  std::array<Direction<VolumeDim>, VolumeDim> result;
+  for (size_t i = 0; i < VolumeDim; i++) {
+    const auto side =
+        mapped(Direction<VolumeDim>(i, Side::Upper)).side() == Side::Upper
```

Comment:
Isn't this literally just

```cpp
const auto side = mapped(Direction<VolumeDim>(i, Side::Upper)).side();
```
? In which case I'd say just inline that call on the `gsl::at` line below.

Reply:
and isn't `mapped(Direction<VolumeDim>(i, Side::Upper))` just `gsl::at(mapped_directions_, i)` ?